### PR TITLE
Update to new interop packages for compat with VS main

### PR DESCRIPTION
**Pull request by debonte**
## Bug

Fixes: NuGet/Client.Engineering#754

Regression? No

## Description
DTE.Window.HWnd changed from returning an int to an IntPtr in VS's main branch. This PR updates the release-6.0.x branch to use newer interop assemblies with this breaking change in preparation for an insertion.

The VS threading analyzers also flagged several new issues. I'm guessing this is because I had to upgrade the Microsoft.VisualStudio.Threading package. The errors are all RunAsync and StartOnIdle calls that were not awaited. I fixed these by:
1. Changed the RunAsync calls to Run calls, but it's possible that in some of these cases we were intentionally not blocking. I'm not sure.
2. Called FileAndForget on the JoinableTasks returned by the StartOnIdle calls. FileAndForget requires a telemetry fault event name to be passed in. That fault will be recorded if the task fails. I'm not sure what format NuGet uses for telemetry event names though. Should these be something along the lines of vs/nuget/featureName/blah?

## PR Checklist

- [X] PR has a meaningful title
- [X] PR has a linked issue.
- [X] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [X] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [X] N/A

### CODE REVIEW

File: src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsoleKeyProcessor.cs [hunk 1]
```diff
@@ -360,7 +360,7 @@ protected override int InternalExec(ref Guid pguidCmdGroup, uint nCmdID, uint nC
                                 }
                                 else
                                 {
-                                    NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate { await TriggerCompletionAsync(); });
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsoleKeyProcessor.cs:363]: I think the safer approach is to not change these and just suppress the warning and get it handled later.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs [hunk 2]
```diff
@@ -556,7 +556,7 @@ public DisplayVersion SelectedVersion
                             PackageMetadata = detailedPackageMetadata;
                         }
 
-                        NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(() => SelectedVersionChangedAsync(_searchResultPackage, _selectedVersion.Version, loadCts.Token).AsTask());
+                        NuGetUIThreadHelper.JoinableTaskFactory.Run(() => SelectedVersionChangedAsync(_searchResultPackage, _selectedVersion.Version, loadCts.Token).AsTask());
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: This could have perf side-effects. I'd suppress it.

File: src/NuGet.Clients/NuGet.Console/Xamls/ConsoleContainer.xaml.cs [hunk 3]
```diff
@@ -58,7 +58,7 @@ public ConsoleContainer()
                             RootLayout.Children.Add(new PackageRestoreBar(_solutionManager, packageRestoreManager));
                             RootLayout.Children.Add(new RestartRequestBar(deleteOnRestartManager, shell));
                         });
-                }, VsTaskRunContext.UIThreadIdlePriority);
+                }, VsTaskRunContext.UIThreadIdlePriority).FileAndForget("ConsoleContainer");
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.Console/Xamls/ConsoleContainer.xaml.cs]: Use `PostOnFailure`. 

Example: https://github.com/NuGet/NuGet.Client/blob/e89974978e0e519657e65a24578afb1d20bf49d9/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L217

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/DetailControl.xaml.cs [hunk 4]
```diff
@@ -72,7 +72,7 @@ public void ScrollToHome()
 
         public void Refresh()
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/DetailControl.xaml.cs]: This would make the refreshing of the PM UI block on the UI thread.

**Review by nkolev92 — COMMENTED** [review]
I think my feedback on RunAsync to Run applies to all instances. 

I think it's be safer/faster for you to suppress the warnings from the threading analyzers for now, and create a follow up issue that can be handled at a later point.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs (hunk 2 above)
Comment (zivkan) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: I'd use `.PostOnFault(..)` on all instances, rather than suppressing it. As I mentioned in the data meeting on Monday, we're not capturing real customer impact issues because of these not-awaited async methods.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProductUpdateBar.xaml.cs [hunk 5]
```diff
@@ -63,7 +63,7 @@ private void OnUpdateLinkClick(object sender, RoutedEventArgs e)
             NuGetUIThreadHelper.JoinableTaskFactory.StartOnIdle(() =>
             {
                 _productUpdateService.Update();
-            });
+            }).FileAndForget("OnUpdateLinkClick");
```
Comment (zivkan) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProductUpdateBar.xaml.cs]: `.PostOnFault`

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs (hunk 2 above)
Comment (debonte) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: I switched to PostOnFailure for all of these calls. But the PostOnFailure method is one of the offending methods. That one I've suppressed. Is there something better we can do there? If not, I'll close the issue I created -- https://github.com/NuGet/Client.Engineering/issues/956

File: build/packages.targets [hunk 6]
```diff
@@ -3,7 +3,7 @@
         <MicrosoftBuildPackageVersion Condition="'$(MicrosoftBuildPackageVersion)' == ''">16.8.0</MicrosoftBuildPackageVersion>
         <NewtonsoftJsonPackageVersion Condition="$(NewtonsoftJsonPackageVersion) == ''">9.0.1</NewtonsoftJsonPackageVersion>
         <SystemPackagesVersion>4.3.0</SystemPackagesVersion>
-        <VSFrameworkVersion>17.0.0-preview-2-31221-277</VSFrameworkVersion>
+        <VSFrameworkVersion>17.0.0-previews-1-31318-023</VSFrameworkVersion>
```
Comment (zkat) [inline build/packages.targets:6]: Was this change intentional?

Comment (debonte) [reply build/packages.targets:6]: Yes, that's really the crux of this change -- upgrade to a VS interop package version compatible with VS's main branch.

Btw the "previews" token (vs. "preview") is correct. Here's how the IDE team explained it:

> Due to our releasing a partner preview or two before our first public preview, our VS SDK nuget packages built from the VS repo had to apparently downgrade from building preview-2 to preview-1. To address this and satisfy nuget's semver rules to recognize newer from older packages, we added an s to the unstable tag. Please note that previews-* packages are newer than preview-*. So your next VS SDK package update will take you from -preview-2 to -previews-1.

Comment (zivkan) [reply build/packages.targets:6]: 31318 is larger than 31221, so I guess so. Especially since dev17 preview 1 hasn't shipped to customers yet, so it doesn't make sense to be on preview 2 already.

edit: github didn't show me that Erik has already replied 😠

**Review by zivkan — APPROVED** [review]

**Review by zkat — APPROVED** [review]

**Comment by debonte** [comment]
Only E2E failure is SimpleBindingRedirectsClassLibraryReference which was failing before this change.
